code/modules/mod/mod_core.dm

```diff
@@ -355,3 +355,71 @@
 	add_charge(uses_needed * charge_given)
 	balloon_alert(user, "core refueled")
 	return TRUE
+
+/obj/item/mod/core/plasma/lavaland
+	name = "MOD plasma flower core"
+	icon_state = "mod-core-plasma-flower"
+	desc = "A strange flower from the desolate wastes of lavaland. It pulses with a strange purple glow.  \
+		The wires coming out of it could be hooked into a MODsuit."
+	light_system = MOVABLE_LIGHT
+	light_color = "#cc00cc"
+	light_range = 2
+
+	/// The mob to be spawned by the core
+	var/mob/living/spawned_mob = /mob/living/basic/butterfly/lavaland/temporary
+	/// Max number of mobs it can spawn
+	var/max_spawns = 3
+	var/can_spawn = TRUE
+	/// Mob spawner for the core
+	var/datum/component/spawner/mob_spawner
```

Comment:
Newly defined vars should be at the bottom, in this case ``maxcharge`` and ``charge`` should be above these.